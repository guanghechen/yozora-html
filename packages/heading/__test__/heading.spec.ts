import type { Emphasis, Heading, Strong, Text } from '@yozora/ast'
import { EmphasisType, HeadingType, StrongType, TextType } from '@yozora/ast'
import { renderChildren } from '../../../jest.setup'
import renderHeading from '../src'

describe('snapshot', function () {
  it('basic', function () {
    const node: Heading = {
      type: HeadingType,
      depth: 1,
      children: [
        {
          type: TextType,
          value: 'yozora is cool!',
        } as Text,
      ],
    }
    expect(renderHeading(node, renderChildren)).toMatchSnapshot()
  })

  it('basic-2', function () {
    const node: Heading = {
      type: HeadingType,
      depth: 6,
      identifier: 'waw',
      children: [
        {
          type: EmphasisType,
          children: [
            {
              type: TextType,
              value: 'Hello',
            } as Text,
          ],
        } as Emphasis,
        {
          type: TextType,
          value: ', ',
        } as Text,
        {
          type: StrongType,
          children: [
            {
              type: TextType,
              value: 'World',
            } as Text,
          ],
        } as Strong,
      ],
    }
    expect(renderHeading(node, renderChildren)).toMatchSnapshot()
  })
})
