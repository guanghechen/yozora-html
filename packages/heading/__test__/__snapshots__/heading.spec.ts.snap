// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot basic 1`] = `"<h1 className=\\"yozora-heading\\"><p className=\\"yozora-heading__content\\"><span class=\\"yozora-text\\">yozora is cool!</span></p></h1>"`;

exports[`snapshot basic-2 1`] = `"<h6 id=\\"waw\\" className=\\"yozora-heading yozora-heading--toc\\"><p className=\\"yozora-heading__content\\"><em class=\\"yozora-emphasis\\"><span class=\\"yozora-text\\">Hello</span></em><span class=\\"yozora-text\\">, </span><strong class=\\"yozora-strong\\"><span class=\\"yozora-text\\">World</span></strong></p><a className=\\"yozora-heading__anchor\\" href=\\"#waw\\"><em class=\\"yozora-emphasis\\"><span class=\\"yozora-text\\">Hello</span></em><span class=\\"yozora-text\\">, </span><strong class=\\"yozora-strong\\"><span class=\\"yozora-text\\">World</span></strong></a></h6>"`;
